MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs

```diff
@@ -7077,5 +7087,80 @@ private void ul2as()
         ///     Signature: int errcod;
         /// </summary>
         private ReadOnlySpan<byte> errcod => Module.Memory.GetVariablePointer("ERRCOD").Data;
+
+        /// <summary>
+        ///     Determines the profanity level of a string
+        ///
+        ///     MBBSEmu doesn't support multiple profanity levels, so the default value of 0 is returned.
+        /// 
+        ///     Signature:  int profan(char *string);
+        /// </summary>
+        private void profan()
+        {
+            Registers.AX = 0;
+        }
+
+        /// <summary>
+        ///     Expect a YES or NO from the user
+        ///
+        ///     Signature: int yesno=cncyesno();
+        /// </summary>
+        private void cncyesno()
+        {
+            //Get Input
+            var inputLength = Module.Memory.GetWord("INPLEN");
+            var inputPointer = Module.Memory.GetVariablePointer("INPUT");
+            var inputString = Module.Memory.GetArray(inputPointer, inputLength);
+            var inputStringComponents = Encoding.ASCII.GetString(inputString).ToUpper().Split('\0');
+
+            if (inputStringComponents.Length == 0)
+            {
+                Registers.AX = 0;
+                return;
+            }
+
+            switch (inputStringComponents[0])
+            {
+                case "YES":
+                case "Y":
+                    Registers.AX = 'Y';
+                    break;
+                case "NO":
+                case "N":
+                    Registers.AX = 'N';
+                    break;
+                default:
+                    Registers.AX = 0;
+                    return;
+            }
+
+            Module.Memory.SetPointer("NXTCMD", new IntPtr16(inputPointer.Segment, (ushort)(inputPointer.Offset + inputStringComponents[0].Length + 1)));
+        }
+
+
+        /// <summary>
+        ///     Converts all uppercase letters in the string to lowercase
+        ///
+        ///     Result buffer is 1k
+        ///
+        ///     Signature: char *lower=strlwr(char *string);
+        /// </summary>
+        private void strlwr()
+        {
+            var inputString = GetParameterString(0, false);
+
+            if (inputString.Length > 0x400)
+            {
+                _logger.Error("Input String is larger than the result buffer. String is being truncated to 1k");
+                inputString = inputString.Substring(0, 0x399) + '\0';
+            }
```

Comment:
we need a test case for this :)

Reply:
Updated unit test. Also 0x399 != 0x3FF 🤦